CPAC/utils/utils.py

```diff
@@ -2657,3 +2657,192 @@ def flip_orientation_code(code):
     """Reverts an orientation code by flipping R↔L, A↔P, and I↔S."""
     flip_dict = {"R": "L", "L": "R", "A": "P", "P": "A", "I": "S", "S": "I"}
     return "".join(flip_dict[c] for c in code)
+
+
+def get_fmap_type(metadata):
+    """Determine the type of field map from metadata.
+
+    reference: https://bids-specification.readthedocs.io/en/latest/modality-specific-files/magnetic-resonance-imaging-data.html#case-1-phase-difference-map-and-at-least-one-magnitude-image
+
+    Parameters
+    ----------
+    metadata : dict or str
+        Metadata dictionary or path to a JSON file containing metadata.
+
+    Returns
+    -------
+    str or None
+        Returns the type of field map as a string:
+        - "phasediff" for phase difference maps with two echo times
+        - "phase" for single echo phase maps
+        - "fieldmap" for field maps with units like Hz, rad/s, T, or Tesla
+        - "epi" for EPI field maps with phase encoding direction
+    """
+    if not isinstance(metadata, dict):
+        if isinstance(metadata, str) and ".json" in metadata:
+            import json
+
+            try:
+                with open(metadata, "r", encoding="utf-8") as f:
+                    metadata = json.load(f)
+            except (FileNotFoundError, json.JSONDecodeError):
+                return None
+        else:
+            return None
+
+    # Check for required BIDS fields only
+    match (
+        "EchoTime1" in metadata,
+        "EchoTime2" in metadata,
+        "EchoTime" in metadata,
+        "Units" in metadata,
+        "PhaseEncodingDirection" in metadata,
+    ):
+        case (True, True, _, _, _):
+            # Case 1: Phase-difference map (REQUIRED: EchoTime1 AND EchoTime2)
+            return "phasediff"
+        case (False, False, True, _, False):
+            # Case 2: Single phase map (REQUIRED: EchoTime, but NOT EchoTime1/2)
+            return "phase"
+        case (_, _, _, True, _):
+            # Case 3: Direct field mapping (REQUIRED: Units)
+            units = metadata["Units"].lower()
+            if units in ["hz", "rad/s", "t", "tesla", "hertz"]:
+                return "fieldmap"
```

Comment:
If `"Units"` is in `metadata` but `metadata["Units"]` isn't any of these, we want to return `None`? If so, I think it would be clearer to include it here

```suggestion
        case (_, _, _, True, _):
            # Case 3: Direct field mapping (REQUIRED: Units)
            units = metadata["Units"].lower()
            if units in ["hz", "rad/s", "t", "tesla", "hertz"]:
                return "fieldmap"
            return None
```

instead of after the `match` block (move the line from https://github.com/FCP-INDI/C-PAC/blob/cf9229c51f6d41a5d6e163e802d08c527f950eaf/CPAC/utils/utils.py#L2718 into the block; we don't need a `return` outside the block if every condition in the block returns.)